csharp/rust/src/data.rs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+use crate::value::Value;
+use std::ffi::{
+    c_int, c_void,
+};
+use std::fmt::{Display, Formatter};
+use std::os::raw::c_char;
+use std::ptr::null;
+use std::str::Utf8Error;
+use tokio::runtime::Runtime;
+
+pub struct FFIHandle {
+    pub runtime: Runtime,
+    pub handle: crate::apihandle::Handle,
+}
+
+#[repr(C)]
+#[derive(PartialEq)]
+pub struct CreateClientHandleResult {
+    pub result: ECreateClientHandleCode,
+    pub client_handle: *const c_void,
+    pub error_string: *const c_char,
+}
+
+#[repr(C)]
+#[derive(PartialEq, Eq, Debug, Clone, Copy, Hash)]
+pub enum ECreateClientHandleCode {
+    Success = 0,
+    ParameterError = 1,
+    ThreadCreationError = 2,
+    ConnectionTimedOutError = 3,
+    ConnectionToFailedError = 4,
+    ConnectionToClusterFailed = 5,
+    ConnectionIoError = 6,
+}
+
+#[repr(C)]
+pub struct NodeAddress {
+    pub host: *const c_char,
+    pub port: u16,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Utf8OrEmptyError {
```

Comment:
You shouldn't have UTF8 err, since we will marshal `Value::BulkString` as a `byte[]` and won't create a string from it [in rust].
Please remove

Reply:
This is for being panic safe. There are, eg., such potential errors when passing the address on connection request. This being a helper method, will otherwise cause program aborts when utf errors are happening on invalid usage.